### DISCUSSION TITLE
vela cli plugin for asdf

### DISCUSSION THREAD

**Discussion by pdemagny** [kubevela/kubevela#5114] [Ideas] [answered]
Hi,

Much like I did with [liqoctl](https://github.com/liqotech/liqo/discussions/1546), I made a plugin for [asdf](https://github.com/asdf-vm/asdf) to install the vela CLI easily on Linux/MacOS hosts with this great tool.

I would like to know if:
1. You are ok with this, my repo can stay public, and you give me the permissions to make a PR to make my plugin (which is based on the official asdf plugin template) a permanent plugin of asdf by creating a PR on [the asdf plugins repo](https://github.com/asdf-vm/asdf-plugins). Optionally, I could also make a PR in [the kubevela.io repository](https://github.com/kubevela/kubevela.io) to update the installation docs with this new installation method.
2. You are not ok with this and I shall make my repo private and not submit it to the asdf plugins repo !

Please tell me what you think ;)

ps: [the link to the plugin](https://github.com/pdemagny/asdf-vela)

**Comment by wonderflow** [comment, marked as answer]
@pdemagny We're glad to see this happen, thanks! Please just do that and add the installation docs for kubevela. I think our users would also appreciate for your contributions!

**Reply by wonderflow** [reply]
BTW, I think velad can also be a proper option for asdf installation, the velad tool can launch a k3s based vela control plane along with the CLI. https://github.com/kubevela/velad

**Reply by pdemagny** [reply]
Hi, I've created both [vela](https://github.com/pdemagny/asdf-vela) and [velad](https://github.com/pdemagny/asdf-velad) plugins for asdf-vm, I made the [vela](https://github.com/asdf-vm/asdf-plugins/pull/709) and [velad](https://github.com/asdf-vm/asdf-plugins/pull/710) PRs to the asdf-plugins repo, and the [documentation PR on kubevela.io repo](https://github.com/kubevela/kubevela.io/pull/1083) !
